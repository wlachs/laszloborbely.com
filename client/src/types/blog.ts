export type BlogPost = {
	urlHandle: string;
	title: string;
	author: string;
	summary: string;
	body: string;
	creationTime: Date;
};
