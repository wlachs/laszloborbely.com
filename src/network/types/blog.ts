export type BlogPostData = {
	author: string;
	body: string;
	creationTime: Date;
	id: string;
	summary: string;
	title: string;
};
