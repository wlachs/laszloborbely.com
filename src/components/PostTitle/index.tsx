import {type BlogPostData} from '../../network/types/blog.ts';

function PostTitle({post}: {
	post: BlogPostData;
}) {
	return (
		<h1>{post.title}</h1>
	);
}

export default PostTitle;
