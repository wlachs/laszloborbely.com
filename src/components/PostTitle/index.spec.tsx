import {cleanup, render, screen} from '@testing-library/react';
import {afterEach, describe, expect, it} from 'vitest';
import {type BlogPostData} from '../../network/types/blog.ts';
import PostTitle from './index.tsx';

const post: BlogPostData = {
	urlHandle: 'test-post',
	title: 'very very long test post',
	author: 'me myself and I',
	summary: 'a summery summary',
	body: 'body with a six-pack of beer',
	creationTime: new Date(0),
};

describe('PostTitle component', () => {
	afterEach(() => {
		cleanup();
	});

	it('should match snapshot', () => {
		const {container} = render(
			<PostTitle post={post}/>,
		);
		expect(container).toMatchSnapshot();
	});

	it('should display post title', () => {
		render(
			<PostTitle post={post}/>,
		);
		expect(screen.getByText(post.title)).toBeDefined();
	});
});
