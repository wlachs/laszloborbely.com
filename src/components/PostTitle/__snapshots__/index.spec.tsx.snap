// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostTitle component > should match snapshot 1`] = `
<div>
  <h1>
    very very long test post
  </h1>
</div>
`;
